api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -3693,6 +3695,81 @@ message ElasticsearchRequest {
   string Query = 12 [(gogoproto.jsontag) = "query"];
 }
 
+// DynamoDBRequest is emitted when a user executes a DynamoDB request.
+// We support DynamoDB access via app access and database access,
+// therefore this event includes either DatabaseMetadata or AppMetadata.
+message DynamoDBRequest {
+  // OperationPlane specifies DynamoDB request operation plane.
+  enum OperationPlane {
+    // GENERAL is for otherwise uncategorized operations.
+    GENERAL = 0;
+    // CONTROL is for control-plane operations: create and manage DynamoDB tables.
+    CONTROL = 1;
+    // DATA is for data-plane operations: CRUD operations on data in a table.
+    DATA = 2;
+    // STREAM is for stream operations: enable/disable a stream on a table or access data modification records in a stream.
+    STREAM = 3;
+    // TRANSACTION is for transaction opertions that provide ACID guarantees.
+    TRANSACTION = 4;
+  }
+
+  // Metadata is a common event metadata.
+  Metadata Metadata = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // User is a common user event metadata.
+  UserMetadata User = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // SessionChunkID is the ID of the session that was created for this 5 minute
+  // application log chunk.
+  string SessionChunkID = 3 [(gogoproto.jsontag) = "session_chunk_id"];
```

Comment:
Do we still need session chunk ?

Reply:
Yes. In Database access we emit events to both the audit log and to a session for upload when the session ends. 

For example: 
<img width="1155" alt="image" src="https://user-images.githubusercontent.com/16846283/196264240-e0b700c1-36e5-4344-90a2-b6ac518cf5b9.png">

If I play back the session uploaded you can see all 3 events (start/query/end) are recorded within the session as well:

```
$ tsh play --format json 48d498b4-fc2e-405f-a345-f12386921d95 | jq                                  (base) 
{
  "cluster_name": "gavin-dev.cloud.gravitational.io",
  "code": "TDB00I",
  "db_protocol": "mysql",
  "db_service": "teleport-mysql",
  "db_uri": "teleport-mysql.mysql.database.azure.com:3306",
  "db_user": "teleport50",
  "ei": 0,
  "event": "db.session.start",
  "namespace": "default",
  "server_id": "08cf9ef5-e3be-49ff-9d05-6c8c0cb9e743",
  "sid": "48d498b4-fc2e-405f-a345-f12386921d95",
  "success": true,
  "time": "2022-10-17T00:40:40.084Z",
  "uid": "fb89307f-b033-443e-a446-8054e1bc8532",
  "user": "gavin.frazar@goteleport.com"
}
{
  "cluster_name": "gavin-dev.cloud.gravitational.io",
  "code": "TDB02I",
  "db_protocol": "mysql",
  "db_query": "select @@version_comment limit 1",
  "db_service": "teleport-mysql",
  "db_uri": "teleport-mysql.mysql.database.azure.com:3306",
  "db_user": "teleport50",
  "ei": 1,
  "event": "db.session.query",
  "sid": "48d498b4-fc2e-405f-a345-f12386921d95",
  "success": false,
  "time": "2022-10-17T00:40:40.146Z",
  "uid": "1a74c7b1-dc63-476a-b858-345821b4d1f1",
  "user": "gavin.frazar@goteleport.com"
}
{
  "cluster_name": "gavin-dev.cloud.gravitational.io",
  "code": "TDB01I",
  "db_protocol": "mysql",
  "db_service": "teleport-mysql",
  "db_uri": "teleport-mysql.mysql.database.azure.com:3306",
  "db_user": "teleport50",
  "ei": 2,
  "event": "db.session.end",
  "sid": "48d498b4-fc2e-405f-a345-f12386921d95",
  "time": "2022-10-17T00:40:41.181Z",
  "uid": "1ba7642c-f3d0-4809-83f4-6a99e019725f",
  "user": "gavin.frazar@goteleport.com"
}
```

Reply:
I can't think of a better way to define a "session" in app access. I realized it would be best to just keep the definition as a 5 minute TTL "chunk" of data.

One thing that you may have noticed is that `AppSessionRequest` does not include the session chunk id nor the session id.  This is because in the Web UI you do not see `AppSessionRequest` - you can only see them by playing back a session chunk.

For DynamoDB, since we want to see each request in the WebUI, I included the session chunk ID so that users can associate the log entry with the session chunk it is included in. This way, they can play back the associated chunk if they want to. Otherwise, determining which chunk a dynamo query came from would be awkward and difficult.

Reply:
Yes but in database access I don't think that  chunkID is present in `db.session.query` event in the current flow  for `tsh proxy aws` the [AppSessionRequest](https://github.com/gravitational/teleport/blob/a75fcc21d87c6d06ba7d8b358150e5a78d55e0f3/lib/srv/app/aws/handler.go#L165)  doesn't contain chunkID.

Reply:
Reposting here since we discussed this in slack: db start/end/query events all contain the session id, which is used to play back a db session. For app access, it's the session chunk id that is used for session playback. Therefore we are keeping the chunk id in the event for better UX.